### ISSUE TITLE
HC enables SmartScreen on every launch

### ISSUE THREAD

**Issue by YZgitter** [AndyFul/Hard_Configurator#18]
If SmartScreen is disabled, Hard Configurator enables it when it's launched. This occurs even if 'Forced SmartScreen' is set to 'OFF'.

**Comment by AndyFul** [comment]
That is an intended behavior. Any sensible configuration in H_C requires enabled SmartScreen. 
"Forced SmartScreen" set to OFF does not disable SmartScreen.

**Comment by YZgitter** [comment]
> "Forced SmartScreen" set to OFF does not disable SmartScreen.

Sure, and I wasn't implying that. If the user sets 'Forced SmartScreen' to ON, enabling SmartScreen would make some sense. If it's OFF, not so much.

If SmartScreen is disabled on a PC, it's either because of
1. a user preference or
2. a problem, maybe even malware

Both situations should be considered and respected. If it's 1, let them use their PC as they want, with SmartScreen disabled. If it's 2, tell them to fix the problem. At most, ask to enable it with a yes/no dialog.

**Comment by AndyFul** [comment]
I do not agree. When installing the security-oriented application, the user should accept that it installs and configures the needed resources. Asking is not a good idea, because the user does not know how exactly H_C works. 
So, I could add the alert with two options:
1. "Please enable SmartScreen and continue" 
2. "Exit the installation"
This would irritate 99% of users.
So, I prefer to show the alert that SmartScreen is enabled. Only 1% of users will be irritated, instead of 99%.

**Comment by YZgitter** [comment]
> configures the needed resources

But SmartScreen is not needed? I have it disabled and HC works fine and does everything that I want.

**Comment by AndyFul** [comment]
It is OK. You can use H_C in a way that was not intended by the author. If you want, we can discuss your settings on the Malwaretips forum:
https://malwaretips.com/threads/hard_configurator-windows-hardening-configurator.66416/

**Comment by YZgitter** [comment]
> It is OK. You can use H_C in a way that was not intended by the author.

First of all, this is about SmartScreen, not HC. SmartScreen is not required for HC to work properly, except for the completely optional 'Forced SmartScreen' feature.
So it's more that I want to use _my computer_ in a way that I like. Maybe you think SmartScreen is good and you want to recommend it to people. That's fine, but why force it onto everyone like that?

And you said 90% of users would be annoyed if HC asked instead of enabling SmartScreen automatically. But 90% of users would never see that question in the first place, because they have SmartScreen enabled already!
The only ones who get annoyed are those who want SmartScreen disabled and are currently forced to manually disable it again after every HC launch. Do you see what I mean and why this is an issue?

Maybe you can at least add a command line argument that stops this behavior?

**Comment by AndyFul** [comment]
As I said, the H_C was created by me as an application that uses SmartScreen as a crucial part of security. Without enabled SmartScreen, most of the H_C manual would not make sense.

Disabling SmartScreen by a command line could be a kind of solution, but it could be also used by the attackers to decrease the security level of users who rely on SmartScreen. 

I think that in your situation you could create a script that runs H_C and automatically disables SmartScreen after closing H_C.

**Comment by AndyFul** [comment]
If my idea of H_C was not fixed on SmartScreen, I could add a few years ago an additional option < Enable SmartScreen >. 
If < Enable SmartScreen > = OFF, then SmartScreen would not be automatically enabled when running H_C.
But when creating H_C, my intentions were fixed on using SmartScreen. Currently, it would be hard to implement that option without much work on help files, H_C manual, settings backups, etc.
So, H_C will be a camel and not a camel with removable humps. :)

**Comment by YZgitter** [comment]
You've made it clear that you think SmartScreen should be enabled, because you think it's a good security feature. That's fine, I don't want to argue about that.

But SmartScreen is enabled by default in Windows, and disabling it takes some effort. So 95% or more of users already have it enabled!
The only thing this HC behavior is doing is annoying the few of us who _don't want_ to have it enabled.
Not everyone has the same priorities and the same security requirements. And I think you do understand that, that's why you added many settings in HC. That's why I don't understand forcing SmartScreen to _people who don't want it_.

And I'm definitely not asking for a SmartScreen ON/OFF switch in Hard_Configurator, that would be stupid and pointless. All I'm asking is to not change the Windows settings that have been set by the user. (Or at least ask before doing that.)

**Comment by AndyFul** [comment]
I am sorry.
